internal/loggerfx/loggerfx.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2023 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package loggerfx
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/goschtalt/goschtalt"
+	"github.com/xmidt-org/sallust"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxevent"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	deaultConfigPath = "logging"
+)
+
+var (
+	ErrInvalidConfigPath = errors.New("configuration structure path like 'logging' or 'foo.logging' is required")
+)
+
+// Module is function that builds the loggerfx module based on the inputs.  If
+// the configPath is not provided then the default path is used.
+func Module(configPath ...string) fx.Option {
+	configPath = append(configPath, deaultConfigPath)
+
+	var path string
+	for _, cp := range configPath {
+		if cp != "" {
+			path = cp
+			break
+		}
+	}
+
+	// Why not use sallust.WithLogger()?  It is because we want to provide the
+	// developer configuration based on the sallust configuration, not the zap
+	// options.  This makes the configuration consistent between the two modes.
+	return fx.Options(
+		// Provide the logger configuration based on the input path.
+		fx.Provide(
+			goschtalt.UnmarshalFunc[sallust.Config](path),
+			provideLogger,
+		),
+
+		// Inform fx that we are providing a logger for it.
+		fx.WithLogger(func(log *zap.Logger) fxevent.Logger {
+			return &fxevent.ZapLogger{Logger: log}
+		}),
+	)
+}
+
+// DefaultConfig is a helper function that creates a default sallust configuration
+// for the logger based on the application name.
+func DefaultConfig(appName string) sallust.Config {
```

Comment:
This is a good convention, but it should stay a convention.  I'm ok with this being in `skeleton`, but we should express a dependency on this library via `go.mod`